### PR TITLE
check duplicate attr before create schema.

### PR DESCRIPTION
**Pull request by Susheldon**
check duplicate attr before create schema.

Signed-off-by: xxxxs Susheldon@users.noreply.github.com

### CODE REVIEW

File: geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/GeoMesaSchemaValidator.scala [hunk 1]
```diff
@@ -120,3 +121,13 @@ object IndexConfigurationCheck {
     require(sft.getAttributeShards > 0 && sft.getAttributeShards < 128, "Attribute shards must be between 1 and 127")
   }
 }
+
+object PropertyCheck {
+  def validateDuplicateProperty(sft: SimpleFeatureType):Unit = {
+    val names = sft.getAttributeDescriptors.asScala.map(descriptor => descriptor.getLocalName)
+    val duplicateProperties = names.diff(names.distinct)
+    if ( duplicateProperties.nonEmpty ){
+      throw new IllegalArgumentException(s"duplicate attr: '${duplicateProperties.mkString(",")}' ")
```
Comment (elahrvivaz) [inline geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/GeoMesaSchemaValidator.scala:130]: Could you make the error message a little clearer?

```suggestion
      throw new IllegalArgumentException(s"Schema contains duplicate attribute names: '${duplicateProperties.mkString(", ")}' ")
```

**Comment by elahrvivaz** [comment]
Thanks for adding this, could you also add a quick unit test in https://github.com/locationtech/geomesa/blob/main/geomesa-index-api/src/test/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStoreTest.scala?

**Comment by Susheldon** [comment]
> Thanks for adding this, could you also add a quick unit test in https://github.com/locationtech/geomesa/blob/main/geomesa-index-api/src/test/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStoreTest.scala?

Sorry, I’m a little confused, why add unit test in GeoMesaDataStoreTest? You means write a unit test for the entire process of createSchema?

**Comment by elahrvivaz** [comment]
yes, for things like this it's good to have a unit test to verify that the fix is working, and to prevent any regression due to code changes in the future. in this case, it should be pretty simple, i.e. something like:
```
val sft = SimpleFeatureTypes.createType("foo:String,bar:Int,foo:Double")
val ds = new TestGeoMesaDataStore(true)
ds.createSchema(sft) must throwAn[IllegalArgumentException]
```

**Review by elahrvivaz — APPROVED** [review]

**Comment by elahrvivaz** [comment]
merged as https://github.com/locationtech/geomesa/commit/13e53e7bbb3afafc550aed3a9d644056b369c4c9

**Pull request closed without merging**